src/contexts/auth/AuthContext.tsx

```diff
@@ -52,97 +53,102 @@ export const useAuthActions = (): AuthActions => {
+   * Fully logs user out by calling the logout endpoint and logging out in app state
    */
-  const logOut = useCallback(() => {
+  const logOut = useCallback(async () => {
+    await _fetch('/auth/logout', 'logout', { body: {} });
     setLoggedOut();
-    setToken('');
-    setUserId('');
-    setUserSigninAddress('');
-    /**
-     * NOTE: clearing localStorage completely will also clear the SWR cache, which
-     * will require users to re-fetch data if they visit a profile they've already
-     * visited (including their own). In the future, we should clear data more
-     * selectively (such as only sensitive data)
-     */
-    clearLocalStorageWithException([]);
-  }, [setLoggedOut, setToken, setUserId, setUserSigninAddress]);
-
-  const logIn = useCallback(
-    async (payload: LoggedInState, address: string) => {
+  }, [setLoggedOut]);
+
+  const setLoggedIn = useCallback(
+    async (address: string) => {
       try {
-        setToken(payload.jwt);
-        setUserId(payload.userId);
-        setAuthState(payload);
+        setAuthState(LOGGED_IN);
         setUserSigninAddress(address.toLowerCase());
       } catch (error: unknown) {
         const errorMessage = error instanceof Error ? error.message : 'Unknown error';
-        logOut();
+        setLoggedOut();
         throw new Error('Authorization failed! ' + errorMessage);
       }
     },
-    [logOut, setToken, setUserId, setUserSigninAddress]
+    [setLoggedOut, setUserSigninAddress]
   );
 
   const setStateToLoading = useCallback(() => {
     setAuthState(LOADING);
   }, []);
 
-  const { pushToast } = useToastActions();
+  const handleUnauthorized = useCallback(() => {
+    setLoggedOut(true);
+  }, [setLoggedOut]);
+
+  useEffect(() => {
+    setIsLoggedInLocally(authState === LOGGED_IN);
+  }, [authState, setIsLoggedInLocally]);
+
+  useEffect(() => {
+    async function getAuthenticatedUser() {
+      try {
+        const authenticatedUser = await _fetch('/users/get/current', 'get current user');
 
-  useEffect(
-    function authProviderMounted() {
-      async function loadAuthState() {
-        // Show user loading screen while we figure out whether they are logged in or not
-        setAuthState(LOADING);
-
-        if (token && userId && userSigninAddress) {
-          // const response = await _fetch<ValidateJwtResponse>('/auth/jwt_valid', 'validate jwt');
-          // if (!response.valid) {
-          //   pushToast('Your session is invalid or expired. Please try again.');
-          //   logOut();
-          //   return;
-          // }
-
-          await logIn({ jwt: token, userId }, userSigninAddress);
+        if (authenticatedUser) {
+          setAuthState(LOGGED_IN);
           return;
         }
 
-        setLoggedOut();
-      }
+        // if no authenticated user is returned but they were logged in previously, show a toast
+        if (isLoggedInLocally) {
+          pushToast(EXPIRED_SESSION_MESSAGE);
+        }
```

Comment:
this should continue to work because `isLoggedInLocally` will come from persisted state